### PR TITLE
Honor editor: false across cell types, add readOnlyStyling option

### PR DESCRIPTION
**Pull request by sl01k**
### Context

The PR fixes `editor: false` being ignored in the two places users hit first, and adds an option that removes the reason people reach for it by mistake.

`editor: false` was reported as "not working". It turned out to be three separate things:

1. **A column `type` silently re-enabled it.** `editor: false` at grid level plus `columns: [{ type: 'numeric' }]` produced a fully editable column. `extendByMetaType()` asks `hasOwnProperty` before applying a type's editor, and a grid-level setting reaches a column through the prototype chain, so it was invisible. Since `false` names no editor — it disables editing — and no built-in cell type declares one, there is no type *default* competing here, only a user policy the type undid. A type now never supplies its `editor` when the meta already resolves `editor` to `false`. Everything else the type provides still applies, so a non-editable column keeps its renderer and validator.

2. **Checkboxes ignored it entirely.** A `type: 'checkbox'` column with `editor: false` stayed fully clickable. A checkbox has no separate editing gesture — the click *is* the edit — so the box is now rendered `disabled`. The theme already ships a complete `:disabled` token set for it. A matching `pointer-events: none` keeps the cell selectable by clicking it, which the native attribute would otherwise prevent.

3. **People used `editor: false` to get "read-only without the grey".** That is what `readOnly` is for, but its `htDimmed` styling is not separable from it without naming a CSS class. The new **`readOnlyStyling`** option (boolean, default `true`, so nothing changes for anyone) makes a grid read-only while it keeps looking like a normal table. Only the appearance changes: pasting, drag-to-fill, and `aria-readonly` are untouched.

The `radio` recipe carried the same gate as the checkbox renderer (`readOnly` only), so it is fixed the same way, and it no longer swallows the click on a non-editable cell.

**Known limitation, documented and pinned by tests:** the carve-out in (1) acts when a type is expanded, so `updateSettings({ editor: false })` that does not restate `columns` does not reach an already-built typed column. Pass `columns` in the same call.

**Breaking:** (1) and (2) change behavior — a typed column under a grid-level `editor: false` becomes non-editable, and such a checkbox stops toggling. Both are the documented intent of the option; neither breaks an existing test. `readOnlyStyling` is purely additive.

### Test evidence (required for source changes)

- Unit tests added/modified (`*.unit.js`): `handsontable/src/dataMap/metaManager/__tests__/utils.unit.ts`, `handsontable/src/dataMap/metaManager/__tests__/metaManager.unit.ts`, `handsontable/src/renderers/baseRenderer/__tests__/baseRenderer.unit.ts`
- E2E tests added/modified (Playwright `tests/e2e/*.spec.ts`): `tests/e2e/editor-disabled.spec.ts`, `tests/e2e/read-only-styling.spec.ts` (+ fixtures and page objects)
- Type tests (`*.types.ts`) updated if public API changed: `handsontable/src/__tests__/core/settings.types.ts`
- For a bug fix — the spec that fails without this fix: each change was verified by reverting the source and re-running. Cascade: `should not supply the type's editor when editing is disabled on a PROTOTYPE of the meta object` and `should keep editing disabled when a column declares a "type"`. Checkbox: `a checkbox cannot be toggled when the grid has no editor` and `a checkbox with no editor is rendered disabled`. `readOnlyStyling`: `should not add the readOnly class name when 'readOnlyStyling' is 'false'` and `a read-only cell is painted like an editable one when styling is off`.
- Demo page / recorded trace (for UI changes): none committed — verified in a local `dev-pr.html` against the released build.

### Commands run

```bash
npm run test:unit --prefix handsontable
```
exit 0

```bash
npm run test:e2e --prefix handsontable -- --testPathPattern=checkbox
npm run test:e2e --prefix handsontable -- --testPathPattern=renderers
npm run test:e2e --prefix handsontable -- --testPathPattern=filters
```
`82 specs, 0 failures` / `314 specs, 0 failures` / `285 specs, 0 failures`

```bash
cd tests && npx playwright test --project=e2e-main e2e/editor-disabled.spec.ts e2e/read-only-styling.spec.ts
```
`PASS (17) FAIL (0)`

```bash
npm run eslint --prefix handsontable && npm run stylelint --prefix handsontable && npm run test:types --prefix handsontable
npm --prefix docs run docs:api
```
clean (the only remaining eslint findings are pre-existing, in `src/helpers/__tests__/function.unit.ts`, untouched here); `docs:api` exit 0

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):

1. None — spontaneous work, no ClickUp task.
2.
3.

### Affected project(s):

- [x] `handsontable`
- [ ] `@handsontable/angular-wrapper`
- [ ] `@handsontable/react-wrapper`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://cla.handsontable.com/sign) — one signature covers both Handsontable and HyperFormula; the `cla/signed` check on this PR confirms it.
- [x] My change requires a change to the documentation. Done in this PR: the `editor` and new `readOnlyStyling` JSDoc in `metaSchema.ts`, the read-only cells guide, the regenerated option-levels matrix, and `metaManager/AGENTS.md`.
- [ ] MANUAL QA NEEDED — 

#### Reviewer note

`docs/content/recipes/cell-types/radio/javascript/example1.js` and `react/example1.jsx` show ~180 changed lines. Three of them are the fix; the rest is reformatting, because the committed files were produced by an older version of the example generator and `npm run docs:code-examples:generate-js` reformats the whole file. Diffing both with whitespace and comments stripped shows only the intended change.

---

> [!NOTE]
> **Medium Risk**
> Changes core meta cascade and checkbox interaction paths with documented breaking behavior for grids that relied on typed columns re-enabling editors under `editor: false`.
> 
> **Overview**
> Fixes **`editor: false`** being ineffective when a column declares a **`type`** or uses **checkbox** cells, and adds **`readOnlyStyling`** so read-only grids can skip the dimmed look without changing edit restrictions.
> 
> **`editor: false` + cell types:** Type expansion (`extendByMetaType`) no longer applies a type’s `editor` when meta already resolves `editor` to `false` (including inherited grid-level settings). Renderer and validator from the type still apply. Checkbox cells render as **disabled**, ignore click/Space/autofill toggles, and use CSS so the cell stays selectable.
> 
> **`readOnlyStyling`:** New option (default `true`) at grid/column/cell levels. When `false`, read-only cells omit `readOnlyCellClassName` but keep `aria-readonly` and read-only behavior.
> 
> Docs (read-only guide, option matrix, radio recipe using `getCellEditor`), unit tests, and Playwright e2e for both behaviors. Marked **breaking** where typed columns under grid `editor: false` and non-toggleable checkboxes change behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 08b2542170a8abb2386dfe29deef7fb642ec540a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/fixtures/pages/EditorDisabledPage.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    this.bundle = bundle;
+  }
+
+  /**
+   * Navigate to the fixture and wait for every grid to render. The wait is a
+   * real DOM condition — the LAST grid's first cell — never a readiness flag.
+   */
+  async goto(): Promise<void> {
+    await this.page.goto(`/tests/fixtures/demo/editor-disabled.html?theme=${this.theme}&bundle=${this.bundle}`);
+    await expect(this.cell('typed', 0, 0)).toBeVisible();
```
Comment (demtario) [inline tests/fixtures/pages/EditorDisabledPage.ts:38]: **High:** `goto()` asserts cell visibility without `await awaitBundle(this.page)` first. `tests/AGENTS.md` requires it immediately after `page.goto()`: the bare `expect(...).toBeVisible()` inherits the short `expect` timeout while the ~6MB UMD bundle is still loading, which is the documented cause of intermittent `goto()` timeouts (65/111 page objects call it — see `AutoLinkPage.ts`, same `document.write` shape). Add `await awaitBundle(this.page);` after `page.goto(...)`, importing it from `../bundle`.

File: tests/fixtures/pages/ReadOnlyStylingPage.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    this.bundle = bundle;
+  }
+
+  /**
+   * Navigate to the fixture and wait for every grid to render. The wait is a
+   * real DOM condition — the LAST grid's first cell — never a readiness flag.
+   */
+  async goto(): Promise<void> {
+    await this.page.goto(`/tests/fixtures/demo/read-only-styling.html?theme=${this.theme}&bundle=${this.bundle}`);
+    await expect(this.cell('unstyled', 0, 0)).toBeVisible();
```
Comment (demtario) [inline tests/fixtures/pages/ReadOnlyStylingPage.ts:43]: **High:** same missing `await awaitBundle(this.page)` in `goto()` — see the note on `EditorDisabledPage.ts`. Add it after `page.goto(...)` before asserting on the rendered fixture.

**Review by demtario — COMMENTED** [review]
9 inline notes below.

**Off-diff (no single diff line to anchor):**

- **Missing migration guide (High).** The changelog sets `"breaking": true`, and `.ai/DOC-STANDARDS.md` requires a migration-guide step for a breaking change; none exists in this diff. Target file: `docs/content/guides/upgrade-and-migration/migrating-from-18.1-to-19.0/migrating-from-18.1-to-19.0.md` — §1 there is the template for this exact shape (a bug fix that is still an observable behavior change: who is affected, why, before/after). Already flagged by cursor[bot] on `.changelogs/13584.json`, so not re-posted inline; noting it here because it is load-bearing for merge.
- **Test coverage gaps (Medium).**
  - The `editor: null` arm of `isCheckboxToggleable()` (`checkboxRenderer.ts:496`) is untested. E2E covers only `editor: false`, and no `checkboxRenderer` unit test was added, so a mutation dropping the `!== null` check passes the whole suite.
  - The disabled-checkbox tab-order exclusion is untested — only `toBeDisabled()` is asserted, not that `Tab` skips the box.
  - The documented `updateSettings({ editor: false })`-without-`columns` limitation has unit coverage (`metaManager.unit.ts`) but no E2E via the public `updateSettings()` path, and it is not mentioned in the `read-only-cells` guide this PR touches.

File: docs/content/recipes/cell-types/radio/react/example1.jsx [hunk 3]
```diff
@@ -131,54 +143,56 @@ const radioRenderer = rendererFactory(({ instance, td, row, column, value, cellP
...
-  { task: 'MCP server v1 release prep',        priority: 'high',   status: 'in-progress' },
-  { task: 'GitHub triage rotation docs',       priority: 'low',    status: 'done'        },
-  { task: 'shadcn/ui integration recipe',      priority: 'medium', status: 'todo'        },
+  { task: 'Refactor licensing app navigation', priority: 'high', status: 'in-progress' },
+  { task: 'Theme Builder onboarding tour', priority: 'medium', status: 'todo' },
+  { task: 'MCP server v1 release prep', priority: 'high', status: 'in-progress' },
+  { task: 'GitHub triage rotation docs', priority: 'low', status: 'done' },
+  { task: 'shadcn/ui integration recipe', priority: 'medium', status: 'todo' },
 ];
 /* end:skip-in-preview */
```
Comment (demtario) [inline docs/content/recipes/cell-types/radio/react/example1.jsx:153]: **High (regression):** this `/* end:skip-in-preview */` has no matching `/* start:skip-in-preview */` — the PR's reformat dropped it (develop paired them at 134/142). The preview/demo stripper reads paired markers; an orphan `end` breaks its parsing or strips the wrong span. Restore `/* start:skip-in-preview */` immediately before `const data = [` (~line 146). The `.tsx`/`.js`/`.ts`/`.vue` siblings stay correctly paired.

File: tests/e2e/editor-disabled.spec.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    // The numeric renderer's own marker class — proof the type expansion still
+    // ran, and that only its editor was withheld.
+    await grid.expectCellClass('typed', 0, 1, 'htNumeric');
+    // The text column beside it is untouched by the numeric type.
+    await grid.expectCell('typed', 0, 0, 'Alpha');
+  });
+
+  // A checkbox has no separate editing gesture — clicking the box IS the edit — so with no editor
+  // it must not toggle. Before the fix, `editor: false` on a checkbox column had no effect at all.
+  test('a checkbox cannot be toggled when the grid has no editor', async({ page, theme, bundle }) => {
```
Comment (demtario) [inline tests/e2e/editor-disabled.spec.ts:95]: **Medium:** this test is masked by the native `disabled` attribute. A `disabled` input never dispatches a click-driven toggle, so the assertion holds even if the `onClick`/`onChange` `isCheckboxToggleable()` guards (`checkboxRenderer.ts` ~582, ~608) were reverted — it only re-proves `input.disabled`, which the next test already asserts with `toBeDisabled()`. To exercise the handler guards independently, drive a path that does not rely on native `disabled`. The Space-key test (~119) is unaffected — it hits a distinct guard.

File: docs/content/guides/configuration/configuration-option-levels/option-levels.json [hunk 5]
```diff
@@ -1210,6 +1210,18 @@
...
+    {
+      "name": "readOnlyStyling",
+      "category": "Core",
+      "levels": [
+        "grid",
+        "columns",
+        "cells",
+        "cell"
+      ],
+      "since": null,
```
Comment (demtario) [inline docs/content/guides/configuration/configuration-option-levels/option-levels.json:1222]: **Medium:** `readOnlyStyling` has `"since": null`; it should be `"19.0.0"`. This JSON is generated from `@since` JSDoc tags in `metaSchema.ts`, and the `readOnlyStyling` block has no `@since` (every other option added this cycle carries `@since 19.0.0`). `optionLevels.unit.js` never reads `since`, so CI will not catch it. Fix at source — add `@since 19.0.0` to the `readOnlyStyling` JSDoc and regenerate. (Same root cause as cursor[bot]'s `@since` note.)

File: docs/content/recipes/cell-types/radio/javascript/example1.js [hunk 6]
```diff
@@ -3,169 +3,218 @@ import { registerAllModules } from 'handsontable/registry';
...
+
+    instance.setDataAtCell(row, column, newValue);
+    queueMicrotask(() => {
+      const newTd = instance.getCell(row, column);
+
+      newTd?.querySelector(`input[type="radio"][value="${CSS.escape(newValue)}"]`)?.focus();
     });
-    td.appendChild(wrapper);
-    td.style.verticalAlign = 'top';
+  };
```
Comment (demtario) [inline docs/content/recipes/cell-types/radio/javascript/example1.js:76]: **Medium:** calling `instance.getCellEditor(cellProperties)` inside the renderer runs on every cell draw and throws on an unregistered editor name (`getEditor()` in `core.ts`). This is the render-time failure mode this PR's own `checkboxRenderer` JSDoc deliberately avoids — it reads `cellProperties.editor` directly precisely because resolving through `getCellEditor()` at draw time can take the whole grid down on a misconfigured `editor` string. As the officially-taught custom-cell-type recipe, it should model the safe pattern: read `cellProperties.editor` directly, or add a caveat. Repeats across all five framework variants.

File: docs/content/recipes/cell-types/radio/react/example1.jsx [hunk 7]
```diff
@@ -193,9 +207,9 @@ const ExampleComponent = () => {
       afterInit={afterInit}
       licenseKey="non-commercial-and-evaluation"
     >
-      <HotColumn data="task"     type="text"  width={300} />
-      <HotColumn data="priority" type="radio" width={160} options={priorityOptions} />
-      <HotColumn data="status"   type="radio" width={170} options={statusOptions}   />
+      <HotColumn data="task" type="text" width={300} />
+      <HotColumn data="priority" type="radio" width={160} {...{ options: priorityOptions }} />
+      <HotColumn data="status" type="radio" width={170} {...{ options: statusOptions }} />
```
Comment (demtario) [inline docs/content/recipes/cell-types/radio/react/example1.jsx:212]: **Low:** `{...{ options: priorityOptions }}` (and `status` on line 213) is a spread found nowhere else under `docs/content/recipes/**`; the `.tsx` sibling updated in this same PR keeps plain `options={priorityOptions}`. Revert to the plain form for parity with the sibling and the rest of the tree.

File: handsontable/src/renderers/checkboxRenderer/checkboxRenderer.ts [hunk 8]
```diff
@@ -465,6 +473,29 @@ function registerEvents(instance: HotInstance) {
   return eventManager;
 }
 
+/**
+ * Checks whether a checkbox cell can be toggled by the user.
+ *
+ * A checkbox has no separate editing gesture: clicking the box, or pressing space over it, IS the
+ * edit. So a cell that names NO editor must not toggle, or `editor: false` would have no effect at
+ * all on a checkbox column. `readOnly` is answered by the caller's own checks and is deliberately
```
Comment (demtario) [inline handsontable/src/renderers/checkboxRenderer/checkboxRenderer.ts:481]: **Low (comment accuracy):** "core treats `null` and `false` alike as no editor" holds at `getCellEditor`/`isCellEditable`, but not at type-expansion time — `isEditingDisabled()` (`utils.ts`) tests `=== false` only, so grid-level `editor: null` plus a column `type` yields an editable cell while `editor: false` does not. The predicate here is correct; only the "alike" claim overreaches.

File: tests/fixtures/pages/EditorDisabledPage.ts [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+   * the formatted value: `numericFormat` is applied through numbro, which the
+   * plain UMD bundle leaves external, so the rendered number differs across the
+   * bundle axis while the marker class does not.
+   */
+  async expectCellClass(grid: GridName, row: number, col: number, className: string): Promise<void> {
+    await expect(this.cell(grid, row, col)).toHaveClass(new RegExp(`\\b${className}\\b`));
+  }
+
+  /** The text editor's input, shared by every grid on the page. */
+  editor(): Locator {
```
Comment (demtario) [inline tests/fixtures/pages/EditorDisabledPage.ts:100]: **Low:** `editor()` is an unscoped page-wide `.handsontableInput` locator, but this fixture mounts three grids. It works today (one editor open per test on a freshly-navigated page), but the next test that opens a second grid's editor in the same test will trip Playwright strict-mode multi-match. Scope the locator to the grid under test.
